File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.cpp
class Solution {
public:
    long long maximumSubarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp; long long int s=0; int j=0; long long int ans=0;
        for(int i=0;i<nums.size();i++)
        {
           s+=nums[i];
            mp[nums[i]]++;
            while(mp[nums[i]]>1||i-j+1>k)
            {
                s-=nums[j];
                mp[nums[j]]--;
                j++;
            }
            if(i-j+1==k) ans=max(ans,s);
        }
        return ans;
    }
};

File: 2461-maximum-sum-of-distinct-subarrays-with-length-k/2461-maximum-sum-of-distinct-subarrays-with-length-k.java
class Solution {
    public long maximumSubarraySum(int[] nums, int k) {
        Set<Integer> st=new HashSet<>();
        int j=0; long sum=0; long maxi=0;  
        for(int i=0;i<nums.length;i++)
        {
            while(st.contains(nums[i]))
            {
                st.remove(nums[j]);
                sum-=nums[j];
                j++;
            }
            sum+=nums[i];
            st.add(nums[i]);
            while(i-j+1==k)
            {
                maxi=Math.max(maxi,sum);
                st.remove(nums[j]);
                sum-=nums[j];
                j++;
            }
        }
        return maxi;
    }
}